src/functions/casing.ts

```diff
@@ -36,6 +36,15 @@ export const casing: IFunction<ICasingOptions> = (targetVal, opts) => {
     return;
   }
 
+  if (
+    targetVal.length === 1 &&
+    opts.separator !== void 0 &&
+    opts.separator.allowLeading === true &&
+    targetVal.endsWith(opts.separator.char)
```

Comment:
@P0lip `endsWith` seems a bit "weird" to me here especially when targetVal length == 1

Reply:
this is due to `eslint` rule.
I can change it to `targetVal[0]` and disable the rule for this particular line, because it indeed feels weird